src/index.js

```diff
@@ -1,25 +1,36 @@
 /**
  * @param {number[]} candidates - candidate numbers we're picking from.
- * @param {number} remainingSum - remaining sum after adding candidates to currentCombination.
- * @param {number[][]} finalCombinations - resulting list of combinations.
- * @param {number[]} currentCombination - currently explored candidates.
- * @param {number} startFrom - index of the candidate to start further exploration from.
+ * @param {number} target - number we're searching possible sum combinations
+ * @param {number} index - index of the candidate to start further exploration from.
+ * @param {number[][]} combinations - resulting list of combinations.
+ * @param {number[]} current - currently explored candidates.
  * @return {number[][]}
  */
+const combinationSum = (
+  candidates,
+  target,
+  index = 0,
+  current = [],
+  combinations = []
+) => {
+  // Case base
+  if (target < 0) {
+    return;
+  }
 
-const combinationSumRecursive = (
-    
+  if (target === 0) {
+    combinations.push([...current]);
+    return;
   }
-  
-  /**
-   * Backtracking algorithm of finding all possible combination for specific sum.
-   *
-   * @param {number[]} candidates
-   * @param {number} target
-   * @return {number[][]}
-   */
-const combinationSum = (candidates, target) => {
-    return combinationSumRecursive(candidates, target);
+
+  for (let i = index; i < candidates.length; i++) {
+    const currentValue = candidates[i];
+    current.push(currentValue);
+    combinationSum(candidates, target - currentValue, i, current, combinations);
+    current.pop();
```

Comment:
Por que no usar mejor `[...candidates, currentValue]` para evitar el push and pop

Reply:
`current` va almacenando las combinaciones que se van armando, agrega (push) si la suma es menor que el target, elimina (pop) si la suma es mayor que nuestro target. No tendría sentido copiar todos los candidatos allí.

Gracias por el _code review_.

Reply:
Disculpa, tienes razón me refería a `[...current, currentValue]`
es decir
```js
combinationSum(candidates, target - currentValue, i, [...current, currentValue], combinations);
```
para evitar el push and pop